docs/dotnet/dotnet-signalr-metrics.md

```diff
@@ -0,0 +1,68 @@
+# Semantic Conventions for SignalR metrics emitted by ASP.NET Core
+
+**Status**: [Experimental][DocumentStatus]
+
+This document defines semantic conventions for SignalR metrics emitted by .NET components and runtime.
+
+**Disclaimer:** These are initial .NET metric instruments available in .NET 8 but more may be added in the future.
+
+<!-- toc -->
+
+- [Metric: `signalr_http_transport.connection.duration`](#metric-signalr_http_transportconnectionduration)
+- [Metric: `signalr_http_transport.active_connections`](#metric-signalr_http_transportactive_connections)
+
+<!-- tocstop -->
+
+## Metric: `signalr_http_transport.connection.duration`
+
+**Status**: [Experimental][DocumentStatus]
+
+This metric is required.
+
+<!-- semconv metric.signalr.server.connection.duration(metric_table) -->
+| Name     | Instrument Type | Unit (UCUM) | Description    |
+| -------- | --------------- | ----------- | -------------- |
+| `signalr_http_transport.connection.duration` | Histogram | `s` | The duration of connections on the server. [1] |
+
+**[1]:** **TODO: why http_transport? can we report protocol name as dimension and not a metric name?**
+**TODO: we should do signalr.server**
+Corresponding `EventCounter` name is `connections-duration`; Meter name is `Microsoft.AspNetCore.Http.Connections`
+<!-- endsemconv -->
+
+<!-- semconv metric.signalr.server.connection.duration(full) -->
+| Attribute  | Type | Description  | Examples  | Requirement Level |
+|---|---|---|---|---|
+| `signalr.http_transport.status` | string | TODO [1] | `TODO` | Recommended |
+| `signalr.http_transport.transport` | string | TODO [2] | `TODO` | Recommended |
```

Comment:
Porting active comment threads from the word doc:

## Liudmila Molkova
network.transport

July 18, 2023 at 4:43 PM
## James Newton-King
This value can be long polling, web sockets or server sent events. Does network.transport support those values?


If not, I suggest signalr_http_transport.status and signalr_http_transport.transport

July 18, 2023 at 7:50 PM
## Noah Falk
network.transport in otel for protocols like 'tcp' and 'udp' rather than this. I updated it to signalr naming. Look good?

July 19, 2023 at 4:36 AM
## Liudmila Molkova
Sorry, I meant nework.protocol.name and nework.protocol.version 

July 19, 2023 at 10:45 AM
## Liudmila Molkova
I don't really understand why we need specific signalr transport attributes - we should have what we need in otel to represent any transport and any application-level protocol

July 19, 2023 at 10:46 AM
## Noah Falk
Do you think we should use 'network.protocol.name' and then populate it with custom string constants like 'websocket', 'server_sent_events', and 'long_polling'?


'network.transport' says it is for layer 4 protocols in OTel's docs and I know websockets is typically considered a layer 7. I'm not sure what the others would be considered.

July 20, 2023 at 1:43 AM
## Liudmila Molkova
Added signalr.transport (sse and long-polling), websockets are technically an application layer protocol and should go into network.protocol.name.https://github.com/lmolkova/semantic-conventions/blob/dotnet8-metrics-proposed-diff/docs/dotnet/dotnet-signalr-metrics.md#metric-signalrhttpserveractive_connectionsReasoning we need to record both:network.protocol.name = httpsignalr.transport = sse


Totally open to signalr.http.transport attribute name or to adding websockets to the enum

Reply:
added `signalr.http.transport` and websockets to the enum